WordPress/src/main/java/org/wordpress/android/networking/menus/MenusDataModeler.java

```diff
@@ -0,0 +1,270 @@
+
+        MenuModel menu = new MenuModel();
+        menu.menuId = json.optLong(MENU_ID_KEY);
+        menu.name = json.optString(MENU_NAME_KEY);
+        menu.details = json.optString(MENU_DESCRIPTION_KEY);
+        menu.menuItems = menuItemsFromJson(json.optJSONArray(MENU_ITEMS_KEY));
+        menu.locations = menuLocationsFromJson(json.optJSONArray(MENU_LOCATIONS_KEY));
+        return menu;
+    }
+
+    public static JSONObject menuToJson(MenuModel menu) {
+        if (!isValidMenu(menu)) return null;
+
+        try {
+            JSONObject json = new JSONObject();
+            json.put(MENU_ID_KEY, menu.menuId);
+            if (!ListUtils.isEmpty(menu.menuItems)) {
+                JSONArray itemArray = new JSONArray();
+                for (MenuItemModel item : menu.menuItems) {
+                    itemArray.put(menuItemToJson(item));
+                }
+                json.put(MENU_ITEMS_KEY, itemArray);
+            }
+            return json;
+        } catch (JSONException exception) {
+            AppLog.e(AppLog.T.API, "Error serializing MenuModel to JSON: " + exception);
+        }
+
+        return null;
+    }
+
+    /**
+     * To be considered a valid Menu Location a {@link JSONObject} must:
+     * <ul>
+     *     <li>not be null</li>
+     *     <li>contain a name via {@link MenusDataModeler#LOCATION_NAME_KEY}</li>
+     *     <li>contain a default state via {@link MenusDataModeler#LOCATION_DEFAULT_STATE_KEY}</li>
+     * </ul>
+     *
+     * A name is required to uniquely identify the Menu Location (essentially the primary key). A
+     * default state is required because the default state key is unique to Menu Location responses.
+     */
+    public static boolean isValidMenuLocationJson(JSONObject json) {
+        return json != null && json.has(LOCATION_NAME_KEY) && json.has(LOCATION_DEFAULT_STATE_KEY);
+    }
+
+    public static boolean isValidMenuLocation(MenuLocationModel location) {
+        return location != null && !TextUtils.isEmpty(location.name);
+    }
+
+    public static MenuLocationModel menuLocationFromJson(JSONObject json) {
+        if (!isValidMenuLocationJson(json)) return null;
+
+        try {
+            MenuLocationModel location = new MenuLocationModel();
+            location.name = json.getString(LOCATION_NAME_KEY);
+            location.details = json.getString(LOCATION_DESCRIPTION_KEY);
+            location.defaultState = json.getString(LOCATION_DEFAULT_STATE_KEY);
+            return location;
+        } catch (JSONException exception) {
+            AppLog.e(AppLog.T.API, "Error parsing Menu Location JSON: " + exception);
+        }
+
+        return null;
+    }
+
+    public static List<MenuLocationModel> menuLocationsFromJson(JSONArray json) {
+        if (json == null || json.length() <= 0) return null;
+
+        try {
+            List<MenuLocationModel> locations = new ArrayList<>();
+
+            for (int i = 0; i < json.length(); ++i) {
```

Comment:
That's a legacy habit from my C(++) days. Pre-increment doesn't require creating a copy of the variable (for returning the previous value) so it is theoretically faster. With compiler optimization this is no longer an issue with any language, but old habits die hard.